### PR TITLE
clearer error messages

### PR DESCRIPTION
**Pull request by timrach**
Addresses #19 
... i did'nt realize @smwentum is also working on this :disappointed:

### CODE REVIEW

**Comment by smwentum** [comment]
it happens.

**Comment by rizidoro** [comment]
:ok_hand:

File: src/complex.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ Complex.prototype.subtract = function(a) {
         copy.real = copy.real.subtract(a.real);
         copy.imaginary = copy.imaginary.subtract(a.imaginary);
     } else {
-        throw "InvalidArgument";
+        throw "Invalid Argument (" + a.toString() + "): Subtrahend must be of type Complex, Fraction or Integer.";
```
Comment (nicolewhite) [inline src/complex.js]: TIL the word subtrahend :)

**Comment by nicolewhite** [comment]
Awesome! Do you think we should follow the pattern you use [here](https://github.com/nicolewhite/algebra.js/pull/9/files#diff-c633c1823667e8afb64beeeaf8c9d575R68) and do `throw new Error` for consistency?

**Comment by Ginden** [comment]
I think the thrown errors should be instances of `TypeError`.

**Comment by nicolewhite** [comment]
@timrach What do you think? I think `TypeError` makes sense for the cases where the wrong type is passed in, e.g a float. I was also thinking your parser should throw a `SyntaxError` for invalid strings?

**Comment by timrach** [comment]
I think it makes sense to throw typed errors, so I changed the code accordingly:
- **Invalid Argument** is now of type *TypeError*
- **Divide By Zero** and **No Solution** are now of type *EvalError*
- **Errors thrown in the lexer** are now of type *SyntaxError*
- **Errors thrown in the parser** are of the general type *Error* as they indicate semantic errors, not syntactic errors. However I think *SyntaxError* could be more intuitive... what do you think?

**Comment by nicolewhite** [comment]
Awesome!

I think SyntaxError makes sense for the parser. According to [this resource](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error), "SyntaxError
Creates an instance representing a syntax error that occurs while parsing code in `eval()`." While we're using `algebra.parse` and not `eval`, it's the same concept as `eval`: it parses a string into an object, no? So in their example [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval) they have:

```js
eval("2 + 2"); // returns 4
```

And ours is doing the same:

```js
algebra.parse("2 + 2"); // returns 4
```

So again, while it may not be *technically* correct, I think it makes intuitive sense.

**Pull request merged**